packages/utils/babylon-walk/src/scope.js

```diff
@@ -161,18 +167,16 @@ export let scopeVisitor: Visitors<ScopeState> = {
       }
 
       // Register declarations with the scope.
+      let type: BindingType = 'const';
       if (isVariableDeclaration(node)) {
-        for (let decl of node.declarations) {
-          let ids = t.getBindingIdentifiers(node);
-          for (let id in ids) {
-            scope.addBinding(id, decl);
-          }
-        }
-      } else {
-        let ids = t.getBindingIdentifiers(node);
-        for (let id in ids) {
-          scope.addBinding(id, node);
-        }
+        type = node.kind;
+      } else if (isClassDeclaration(node)) {
+        type = 'let';
+      }
+
+      let ids = t.getBindingIdentifiers(node);
+      for (let id in ids) {
+        scope.addBinding(id, node, type);
```

Comment:
I think this is changing the binding from the `VariableDeclarator` to the `VariableDeclaration` node. Was that intentional?

Reply:
No, it wasn't. I missed that `decl` is passed instead of `node`.

But doesn't it make more sense to have 
```diff
 for (let decl of node.declarations) {
-    let ids = t.getBindingIdentifiers(node);
+    let ids = t.getBindingIdentifiers(decl);
     for (let id in ids) {
         scope.addBinding(id, decl);
     }
 }
```

instead? Right now, every declarator is registered for every binding name in the whole declaration.

Reply:
Heh. It may not matter with our current usage then.

Reply:
Fixed